CliFx.Tests/SuggestDirectiveSpecs.cs

```diff
@@ -0,0 +1,288 @@
+        {
+            var builder = new CliApplicationBuilder();
+
+            commandClasses.ToList().ForEach(c =>
+            {
+                var commandType = DynamicCommandBuilder.Compile(c);
+                builder = builder.AddCommand(commandType);
+            });
+
+            return builder.UseConsole(FakeConsole)
+                          .UseFileSystem(NullFileSystem);
+        }
+
+        [Theory]
+        [InlineData(true, 0)]
+        [InlineData(false, 1)]
+        public async Task Suggest_directive_can_be_configured(bool enabled, int expectedExitCode)
+        {
+            // Arrange
+            var application = TestApplicationFactory(_cmdCommandCs)
+                .AllowSuggestMode(enabled)
+                .Build();
+
+            // Act
+            var exitCode = await application.RunAsync(
+                new[] { "[suggest]", "clifx.exe", "c" }
+            );
+
+            // Assert
+            exitCode.Should().Be(expectedExitCode);
+        }
+
+        [Fact]
+        public async Task Suggest_directive_is_disabled_by_default()
+        {
+            // Arrange
+            var application = TestApplicationFactory(_cmdCommandCs)
+                .Build();
+
+            // Act
+            var exitCode = await application.RunAsync(
+                new[] { "[suggest]", "clifx.exe", "c" }
+            );
+
+            // Assert
+            exitCode.Should().Be(1);
+        }
+
+        [Theory]
+        [InlineData("supply all commands if nothing supplied",
+                        "clifx.exe", 0, new[] { "cmd", "cmd02", "parent", "parent list" })]
+        [InlineData("supply all commands that 'start with' argument",
+                        "clifx.exe c", 0, new[] { "cmd", "cmd02" })]
+        [InlineData("supply command options if match found, regardles of other partial matches (no options defined)",
+                        "clifx.exe cmd", 0, new string[] { })]
+        [InlineData("supply nothing if no commands 'starts with' argument",
+                        "clifx.exe m", 0, new string[] { })]
+        [InlineData("supply completions of partial child commands",
+                        "clifx.exe parent l", 0, new[] { "list" })]
+        [InlineData("supply all commands that 'start with' argument, allowing for cursor position",
+                        "clifx.exe cmd", -2, new[] { "cmd", "cmd02" })]
+        public async Task Suggest_directive_suggests_commands_by_environment_variables(string usecase, string variableContents, int cursorOffset, string[] expected)
+        {
+            // Arrange
+            var application = TestApplicationFactory(_cmdCommandCs, _cmd2CommandCs, _parentCommandCs, _childCommandCs)
+                .AllowSuggestMode()
+                .Build();
+
+            // Act
+            var exitCode = await application.RunAsync(
+                new[] { "[suggest]", "--envvar", "CLIFX-{GUID}", "--cursor", (variableContents.Length + cursorOffset).ToString() },
```

Comment:
Do we need a dynamic environment variable or can it always be the same? It's set per-process so there shouldn't be conflicts, right?